executor/aggregate.go

```diff
@@ -718,24 +746,29 @@ func (e *HashAggExec) execute(ctx context.Context) (err error) {
 	}
 }
 
-func (e *HashAggExec) getGroupKey(row chunk.Row) (string, error) {
-	e.groupValDatums = e.groupValDatums[:0]
-	for _, item := range e.GroupByItems {
-		v, err := item.Eval(row)
-		if item.GetType().Tp == mysql.TypeNewDecimal {
-			v.SetLength(0)
+func (e *HashAggExec) getGroupKey(input *chunk.Chunk) (err error) {
+	numRows := input.NumRows()
+	if len(e.groupKeyBuffer) < numRows {
+		for i := len(e.groupKeyBuffer); i < numRows; i++ {
+			e.groupKeyBuffer = append(e.groupKeyBuffer, make([]byte, 10*len(e.GroupByItems)))
 		}
+	}
+	for i := 0; i < numRows; i++ {
```

Comment:
Will this be better:
1. Move this loop before line 751.
2. `make([]byte, 0, 10*len(e.GroupByItems))` in the last loop.